src/components/admin/Billing/TenantOptions.tsx

```diff
@@ -0,0 +1,56 @@
+import { Skeleton } from '@mui/material';
+import AutocompletedField from 'components/shared/toolbar/AutocompletedField';
+import { useZustandStore } from 'context/Zustand/provider';
+import { useIntl } from 'react-intl';
+import {
+    useBilling_setSelectedTenant,
+    useBilling_tenants,
+} from 'stores/Billing/hooks';
+import { SelectTableStoreNames } from 'stores/names';
+import {
+    useBillingTable_setHydrated,
+    useBillingTable_setHydrationErrorsExist,
+} from 'stores/Tables/Billing/hooks';
+import {
+    SelectableTableStore,
+    selectableTableStoreSelectors,
+} from 'stores/Tables/Store';
+
+function TenantOptions() {
+    const intl = useIntl();
+
+    // Billing Store
+    const tenants = useBilling_tenants();
+    const setSelectedTenant = useBilling_setSelectedTenant();
+
+    // Billing Select Table Store
+    const setHydrated = useBillingTable_setHydrated();
+    const setHydrationErrorsExist = useBillingTable_setHydrationErrorsExist();
+
+    const resetBillingSelectTableState = useZustandStore<
+        SelectableTableStore,
+        SelectableTableStore['resetState']
+    >(SelectTableStoreNames.BILLING, selectableTableStoreSelectors.state.reset);
+
+    return tenants.length > 0 ? (
+        <AutocompletedField
+            label={intl.formatMessage({
+                id: 'common.tenant',
+            })}
+            options={tenants}
+            defaultValue={tenants[0]}
+            changeHandler={(_event, value) => {
+                setSelectedTenant(value);
+
+                resetBillingSelectTableState();
```

Comment:
Resetting the select table store state does not reset the hydration-related state properties. Is this intentional?

Also, I am surprised that the `EntityTable` component does not _reliably_ present a loading state when the query changes. The component listens to the `query.loading` state property, but the listener (i.e., `isValidating`) does not always align with the value of the state slice. When paired with the ripple effects of polling, the _billing history_ table never assumes an intermediate, loading state and, instead, presents stale data while waiting for new query data.

Reply:
Yeah - for the same reason as below. Allow for serving cached data. However, we are not using that functionality right now so we could easily just have it set the hydration props in the table as well.

That is by design so that a loading state of the table is not constantly flashed and we can serve up cached data while waiting on new data.